pipelines/toast_ground_sim.py

```diff
@@ -547,415 +162,173 @@ def parse_arguments(comm):
-        # If current and next CES are on different days, insert a break
-        tz = args.timezone / 24
-        start1 = all_ces[i][3]  # MJD start
-        start2 = all_ces[i + 1][3]  # MJD start
-        scan1 = all_ces[i][4]
-        scan2 = all_ces[i + 1][4]
-        if scan1 != scan2 and do_break:
-            breaks.append(nces + i + 1)
-            do_break = False
-            continue
-        day1 = int(start1 + tz)
-        day2 = int(start2 + tz)
-        if day1 != day2:
-            if scan1 == scan2:
-                # We want an entire CES, even if it crosses the day bound.
-                # Wait until the scan number changes.
-                do_break = True
-            else:
-                breaks.append(nces + i + 1)
-
-    nbreak = len(breaks)
-    if nbreak < comm.ngroups - 1:
-        if comm.world_rank == 0:
-            print(
-                "WARNING: there are more process groups than observing days. "
-                "Will try distributing by observation.",
-                flush=True,
-            )
-        breaks = []
-        for i in range(nces - 1):
-            scan1 = all_ces[i][4]
-            scan2 = all_ces[i + 1][4]
-            if scan1 != scan2:
-                breaks.append(nces + i + 1)
-        nbreak = len(breaks)
-
-    if nbreak != comm.ngroups - 1:
-        raise RuntimeError(
-            "Number of observing days ({}) does not match number of process "
-            "groups ({}).".format(nbreak + 1, comm.ngroups)
-        )
-    return breaks
-
-
-@function_timer
-def create_observation(args, comm, all_ces_tot, ices, noise):
+def create_observation(args, comm, all_ces_tot, ices, noise, verbose=True):
     """ Create a TOAST observation.
 
     Create an observation for the CES scan defined by all_ces_tot[ices].
 
     """
-    ces, site, fp, fpradius, detquats, weather = all_ces_tot[ices]
-
-    (
-        CES_start,
-        CES_stop,
-        CES_name,
-        mjdstart,
-        scan,
-        subscan,
-        azmin,
-        azmax,
-        el,
-        season,
-        date,
-    ) = ces
-
-    _, _, site_lat, site_lon, site_alt = site
-
-    totsamples = int((CES_stop - CES_start) * args.samplerate)
+    ces, site, telescope, fp, fpradius, detquats, weather = all_ces_tot[ices]
+    totsamples = int((ces.stop_time - ces.start_time) * args.sample_rate)
 
     # create the TOD for this observation
 
+    if comm.comm_group is not None:
+        ndetrank = comm.comm_group.size
+    else:
+        ndetrank = 1
+
     try:
         tod = TODGround(
             comm.comm_group,
             detquats,
             totsamples,
-            detranks=comm.comm_group.size,
-            firsttime=CES_start,
-            rate=args.samplerate,
-            site_lon=site_lon,
-            site_lat=site_lat,
-            site_alt=site_alt,
-            azmin=azmin,
-            azmax=azmax,
-            el=el,
-            scanrate=args.scanrate,
+            detranks=ndetrank,
```

Comment:
`ndetrank` survived the renaming to more human understandable names

Reply:
I like to prefix integer count variables with `n`. `detrank` is a well-established variable name in TOAST. The alternative would be `detranks` (as is used in the constructor) but, to me, that looks like a container of detranks, not their total number. For more verbosity, we could use `n_detrank` but it looks clunky. Anyway, short variable names are more acceptable when their lifetime is short (fits on a single terminal window).